### PR TITLE
WIP: Joachim

### PR DESCRIPTION
**Pull request by ewjoachim**
Integrating @zebuline's code and a lot of click boilerplate. There's so much stuff going on that splitting into commits is kind of useless (IMHO). Just read the resulting files :)

### CODE REVIEW

**Comment by ewjoachim** [comment]
And, it's not over, still a lot of code to port from North, plus tests (north had a >90% coverage, let's do the same), travis, tox, black, ...

File: west/__init__.py [hunk 1]
```diff
@@ -1 +1,16 @@
...
+        # Package is in the current directory or in the python path
+        from setuptools.config import read_configuration
+        from os import path as _p
+        _conf = read_configuration(
+            _p.join(_p.dirname(_p.dirname(__file__)),
+                    "setup.cfg"))
+        return _conf["metadata"]["version"]
 
+
+__version__ = _extract_version("raincoat")
```
Comment (zebuline) [inline west/__init__.py]: raincoat ?

File: west/runner.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+                self.block_list[-1].append_line(line)
+        self.block_list[-1].close()
+
+    def run(self, db):
+        for block in self.block_list:
+            block.run(db)
+
+    def contains_non_transactional_keyword(self, file_handler):
+        keywords = getattr(
+            settings, 'NORTH_NON_TRANSACTIONAL_KEYWORDS',
```
Comment (zebuline) [inline west/runner.py]: NORTH :)

File: west/west.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+        target_version=target_version,
+        existing_files=files.get_known_fixtures(),
+        sql_tpl=settings.FIXTURES_TPL)
+
+    if version is None:
+        raise ValueError('Can not find fixtures to init the DB.')
+    return version
+
+
+def is_manual_migration(file_handler):
```
Comment (zebuline) [inline west/west.py]: Not exactly the same, but this function is defined also in the `files.py` module.
Is it possible to factorize ?

**Comment by zebuline** [comment]
no tests ?

File: west/__init__.py (hunk 1 above)
Comment (ewjoachim) [reply west/__init__.py]: guess where this was copy/pasted from :D

File: west/runner.py (hunk 2 above)
Comment (ewjoachim) [reply west/runner.py]: aha !

File: west/west.py (hunk 3 above)
Comment (ewjoachim) [reply west/west.py]: Oh, I probably forgot do delete one of those

**Comment by ewjoachim** [comment]
Not yet, but there will be :) (Il be porting your tests, and adding some for my code)

I've splitted things a bit, so I should be able to test with a few less mocks, we'll see.

**Pull request merged**